File: android/app/src/main/kotlin/com/taajsingh/gurbanilens/domain/MockAsr.kt
package com.taajsingh.gurbanilens.domain

/**
 * Test / preview ASR — returns a fixed transcript. Lets the end-to-end
 * voice → transcript → matcher → result flow run on the JVM during
 * `:app:test` without needing libwhisper.so or a model file.
 */
class MockAsr(private val canned: String) : Asr {
    override val isReady: Boolean = true

    override suspend fun transcribe(samples: FloatArray, config: Asr.Config): Asr.Transcript =
        Asr.Transcript(
            text = canned,
            language = config.language,
            durationMs = (samples.size * 1000L) / 16000L,
        )
}

File: ios/GurbaniLens/GurbaniLens/Domain/MockAsr.swift
import Foundation

/// Returns a fixed transcript. Used in SwiftUI previews and as the bring-up
/// fallback when the bundled Whisper model is missing — so the UI can be
/// exercised without a working JNI/C++ path.
///
/// Mirrors `android/.../domain/MockAsr.kt`.
public final class MockAsr: Asr {
    private let canned: String
    private let simulatedMs: Int64

    public init(canned: String = "ik oankaar sat naam karataa purakh", simulatedMs: Int64 = 250) {
        self.canned = canned
        self.simulatedMs = simulatedMs
    }

    public var isReady: Bool { true }

    public func transcribe(_ samples: [Float], config: AsrConfig) async throws -> AsrTranscript {
        // Tiny synthetic delay so the Transcribing state actually paints.
        try? await Task.sleep(nanoseconds: UInt64(simulatedMs) * 1_000_000)
        // gurmukhi="" — MockAsr's canned text is already Latin; the v1
        // display path falls back to `text` when gurmukhi is empty.
        return AsrTranscript(text: canned, gurmukhi: "", language: config.language, durationMs: simulatedMs)
    }
}
